### PR TITLE
adapter refactory

### PR DESCRIPTION
**Pull request by ramon**

### CODE REVIEW

File: src/Adapter/Redis.php [hunk 1]
```diff
@@ -25,56 +25,61 @@ public function __construct(ClientInterface $redis, $prefix = 'circuit_breaker')
     /**
      * @inheritdoc
      */
-    public function setErrorCount(string $service = 'default', int $value = 0): void
+    public function getErrorCount(string $service = 'default'): int
     {
-        $this->redis->set(
-            sprintf('%s.%s.error_count', $this->prefix, $service),
-            $value
+        return (int)$this->redis->get(
```
Comment (pedrofaria) [inline src/Adapter/Redis.php]: You could use the `getKey` method and put the type cast there

File: src/Adapter/Redis.php [hunk 2]
```diff
@@ -25,56 +25,61 @@ public function __construct(ClientInterface $redis, $prefix = 'circuit_breaker')
...
-     * @return int
+     * @inheritdoc
      */
-    private function getKey(string $key): int
+    public function isCircuitBroke(string $service = 'default'): bool
     {
-        if (!$this->redis->exists($key)) {
-            return 0;
-        }
+        $response = $this->redis->exists(
```
Comment (pedrofaria) [inline src/Adapter/Redis.php]: you can change to `return (bool) $this->redis->exists( ... ); `

**Comment by pedrofaria** [comment]
Code for the `isAvailable` method

```php
    /**
     * @inheritdoc
     */
    public function isAvailable(string $service = 'default'): bool
    {
        if ($this->adapter->getErrorCount($service) <= $this->getThreshold($service)) {
            return true;
        }

        if (!$this->adapter->isCircuitBroke()) {
            $this->adapter->setErrorCount(
                $service,
                $this->getThreshold($service) - 1
            );
            return true;
        }
        $this->adapter->breakCircuit($service, $this->getTimeout($service));

        return false;
    }
```

**Review by pedrofaria — APPROVED** [review]
👍

**Comment by ramon** [comment]
Dropped in favor of #5

**Pull request closed without merging**